### PR TITLE
Brandon/feat/truncated list

### PR DESCRIPTION
**Pull request by brandonminch**
## What does this PR do?

Truncated list component to be used to display a set number of items in RecordSelectionList and display an action button to view all results (note: view all w

## Type

- [X] Feature
- [ ] Bug
- [ ] Tech debt

## What are the relevant tickets?

- [SDEV3-1950](https://sprucelabsai.atlassian.net/browse/SDEV3-1950)

## Does this add new dependencies? Is there an installation procedure? (yarn, bower, npm, etc)

Requires updates to RecordSelectionList to be merged and base branch reset to canary: https://github.com/sprucelabsai/workspace.sprucebot-skills-kit/pull/529/files

## Screenshots (if appropriate)

Screen Shot 2019-07-09 at 8 00 56 AM
Screen Shot 2019-07-09 at 8 01 04 AM

## What gif best describes this PR or how it makes you feel?

idk

### CODE REVIEW

File: packages/react-heartwood-components/src/components/TruncatedList/TruncatedList.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import React, { Component, Fragment } from 'react'
+import cx from 'classnames'
+
+import Button from '../Button/Button'
+import EmptyState from '../EmptyState/EmptyState'
+import RecordSelectionList from '../RecordSelectionList/RecordSelectionList'
+
+import type { RecordSelectionListItemProps } from '../RecordSelectionList/RecordSelectionList'
+
+type Props = {
```
Comment (jtini) [inline packages/react-heartwood-components/src/components/TruncatedList/TruncatedList.js]: Please add prop descriptions. I know this isn't explicitly in the A/C's, but I think we should include these descriptions any time we add a new component to facilitate collaboration.

File: packages/react-heartwood-components/src/components/TruncatedList/TruncatedList-story.js [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+						actions={object('actions', [
+							{
+								text: 'Add a location',
+								kind: 'simple'
+							}
+						])}
+					/>
+					<CardBody>
+						<TruncatedListExample
+							header={text('header', 'Locations')}
```
Comment (jtini) [inline packages/react-heartwood-components/src/components/TruncatedList/TruncatedList-story.js]: Can we take `header` out of this example? I wouldn't recommend using it when it's redundant with the card header title. Just wanna make sure this doesn't create confusion.

File: packages/react-heartwood-components/src/components/TruncatedList/TruncatedList.js (hunk 1 above)
Comment (brandonminch) [reply packages/react-heartwood-components/src/components/TruncatedList/TruncatedList.js]: Good call. was an oversight as I was architecting out the props and meant to come back to it.

File: packages/react-heartwood-components/src/components/TruncatedList/TruncatedList-story.js (hunk 2 above)
Comment (brandonminch) [reply packages/react-heartwood-components/src/components/TruncatedList/TruncatedList-story.js]: 👍

**Comment by corbanb** [comment]
@brandonminch just need the branch updated and its a go.

**Pull request merged**